scss/4-elements/_buttons.scss

```diff
@@ -1,71 +1,111 @@
-button,
-.cads-button {
-  @include cads-interactive-element();
+/**
+ * Buttons
+ * Based on: https://github.com/nhsuk/nhsuk-frontend/blob/master/packages/components/button/_button.scss and
+ *   https://github.com/alphagov/govuk-frontend/blob/master/src/govuk/components/button/_index.scss
+ */
 
-  border: solid $cads-border-width-medium $cads-language__primary-button-colour;
-  border-radius: $cads-border-radius;
-  font-size: 1.125rem;
-  font-weight: $cads-font-weight__bold;
-  padding: (rem-to-px($cads-spacing-3) - 1px) $cads-spacing-3;
-  display: inline-block;
+$button-shadow-size: 4px;
+
+@mixin cads-button-defaults() {
+  -webkit-appearance: none;
+  box-sizing: border-box;
+  vertical-align: top;
+  width: auto;
+  text-align: center;
+  text-decoration: none;
+  position: relative;
   cursor: pointer;
-  box-shadow: 0 2px 0 0 $cads-language__button-shadow-colour;
-  outline: none !important;
-  margin: 0 0 $cads-spacing-5 0;
+  outline: none;
+  display: inline-block;
+
+  font-weight: $cads-font-weight__bold;
+  font-size: 1.125rem;
 
   @include cads-media-breakpoint-down(sm) {
     font-size: 1rem;
   }
 
-  &,
-  &__primary {
```

Comment:
Notable change here is moving to longhand selectors instead of `&__primary`. This had a notable bug because we had `<button>` in the parent selector in that is generated classes like `button__secondary` which is not a real element. Removed this for two reasons:

1. To avoid generating unexpected classes as mentioned above
2. Making it possible to do a basic search in Sass files for things like `.cads-button__secondary`

A secondary change is no longer styling `<button>` as a "Button". As these revised button styles have more prominent active states and a raised effect hooking this on to button elements conflicted heavily with places where `<button>` is used separately to "Buttons" e.g. link-buttons.

Reply:
see comment in #231